### PR TITLE
Port archiver integration to C++

### PR DESCRIPTION
**Pull request by PCMan**

### CODE REVIEW

File: src/filemenu.cpp [hunk 1]
```diff
@@ -226,7 +226,7 @@ FileMenu::FileMenu(Fm::FileInfoList files, std::shared_ptr<const Fm::FileInfo> i
                         addAction(action);
                     }
                 }
-                else {
+                else if(archiver->canCreateArchive()){
```
Comment (PCMan) [inline src/filemenu.cpp]: @tsujan the only thing changed is this line. I only show the "Compress" menu item if the archiver supports compression.
Would you like to help take a look at this to see if your archiver setting is correctly loaded?

Comment (PCMan) [reply src/filemenu.cpp]: @tsujan  If the value of `archiver` pointer is null, then the setting is not correctly loaded.

**Comment by tsujan** [comment]
However, this isn't a blocker, IMO. These nice changes can be merged as they are. Other works could be done later.

**Comment by agaida** [comment]
yes, please push the button

**Comment by tsujan** [comment]
Another step closer to independence from libfm, thanks to @PCMan's magic :)

**Pull request merged**